corpus/convert.py

```diff
@@ -172,6 +187,31 @@ def run(self):
                     else segment.recording.name
                 )
                 segment_track = segment.track + 1 if segment.track else 1
+
+                orth = f" {segment.orth.strip()} "
+
+                if self.exclude_non_speech:
+                    for nst in self.non_speech_tokens:
+
+                        def replace_recursive(orthography, token):
+                            pos = orthography.find(f" {token} ")
+                            if pos == -1:
+                                return orthography
+                            else:
+                                orthography = orthography.replace(f" {token} ", " ")
+                                return replace_recursive(orthography, token)
+
+                        orth = replace_recursive(orth, nst)
+
+                if self.remove_punctuation:
+                    for pt in self.punctuation_tokens:
+                        orth = orth.replace(pt, "")
```

Comment:
why do we need a recursive implementation for non-speech tokens but not for punctuation?

Reply:
that is actually quite the good question.. not sure why I implemented this differently.. this shold work without recursion.. Oo

Reply:
maybe we need some test cases 😇

Reply:
hm yes maybe ^^

Chris assigns Chris :D

Reply:
Well now we have a test case and it is failing because the expected output is 
```
116-288045-0000 1 116  0.00 10.65 <d0> AS I APPROACHED THE CITY I HEARD BELLS RINGING AND A LITTLE LATER I FOUND THE STREETS ASTIR WITH THRONGS OF WELL DRESSED PEOPLE IN FAMILY GROUPS WENDING THEIR WAY HITHER AND THITHER
;; LABEL "d0" "default0" "all other segments of category 0"
```
and the actual output is
```
116-288045-0000 1 116  0.00 10.65 <d0> AS I APPROACHED THE CITY I HEARD BELLS RINGING [noise] AND A LITTLE LATER I FOUND THE STREETS ASTIR WITH THRONGS OF WELL DRESSED PEOPLE IN FAMILY GROUPS WENDING THEIR WAY HITHER AND THITHER
;; LABEL "d0" "default0" "all other segments of category 0"
```
It seems the non recursive implementation does not capture two consecutive occurrences of `[noise]` in the corpus. maybe because the last space of the first match coincides with the first space of the last match? 
maybe we can use re.sub or similar to remedy the problem (e.g. define a match only if {nst} is surrounded by [whitespace, begin-, end-of sentence] but then only remove the token.. 

anyway.. _request change_ until test case works :)

Reply:
I would just re-add recursion